x-pack/plugins/observability/public/hooks/use_fetch_indices.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { HttpSetup } from '@kbn/core/public';
+import { useRef } from 'react';
+import { useDataFetcher } from './use_data_fetcher';
+
+export interface UseFetchIndicesResponse {
+  indices: Index[];
+  loading: boolean;
+  error: boolean;
+}
+
+export interface Index {
+  health?: string;
+  status?: string;
+  name: string;
+  uuid?: string;
+  primary?: number | string;
+  replica?: number | string;
+  documents: number;
+  documents_deleted: number;
+  size: string;
+  primary_size: string;
+  isFrozen: boolean;
+  hidden: boolean;
+  aliases: string | string[];
+  data_stream?: string;
+  [key: string]: any;
+}
+
+export function useFetchIndices(): UseFetchIndicesResponse {
+  const hasFetched = useRef<boolean>(false);
+
+  const {
+    data: indices,
+    loading,
+    error,
+  } = useDataFetcher({
+    paramsForApiCall: {},
+    initialDataState: undefined,
+    executeApiCall: async (
+      _: any,
+      abortController: AbortController,
+      http: HttpSetup
+    ): Promise<any> => {
+      try {
+        const response = await http.get<Index>(`/api/index_management/indices`, {
```

Comment:
Isn't the response type `http.get<Array<Index>>`?

Reply:
Good catch